Content.Server/00Citadel/Worldgen/Tools/FastNoise.cs

```diff
@@ -0,0 +1,2970 @@
+﻿// FastNoise.cs
+//
+// MIT License
+//
+// Copyright(c) 2017 Jordan Peck
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+// The developer's email is jorzixdan.me2@gzixmail.com (for great email, take
+// off every 'zix'.)
+//
+
+// Uncomment the line below to swap all the inputs/outputs/calculations of FastNoise to doubles instead of floats
+//#define FN_USE_DOUBLES
+
+#nullable disable
+
+#if FN_USE_DOUBLES
+using FN_DECIMAL = System.Double;
+#else
+using FN_DECIMAL = System.Single;
+#endif
+using System.Runtime.CompilerServices;
+
+namespace Content.Server._00Citadel.Worldgen.Tools;
+
+public class FastNoise
```

Comment:
Yo, lets write a giant class and not tell anyone whjat its supposed to handle

Reply:
This one doesn't belong to me, blame the original author
:HECK: